pybutton/request.py

```diff
@@ -117,9 +123,18 @@ def request_url(secure, hostname, port, path):
         Returns:
             (str) A complete url made up of the arguments.
     '''
+    query = urlencode(query) if query else ''
     scheme = 'https' if secure else 'http'
     netloc = '{0}:{1}'.format(hostname, port)
 
-    return urlunsplit((scheme, netloc, path, '', ''))
-
-__all__ = [Request, urlopen, HTTPError, request, request_url]
+    return urlunsplit((scheme, netloc, path, query, ''))
+
+__all__ = [
+    Request,
+    urlopen,
+    HTTPError,
+    request,
+    request_url,
+    urlparse,
```

Comment:
instead of exporting these functions, you could consider exporting a handy function that encapsulates the behavior you want to expose, which is converting a full URL into a query dict, I imagine.